frontend/src/views/Join.tsx

```diff
@@ -1,154 +1,27 @@
-      connect(socketEndpoint).then(() => {
-        subscribe(subscribeUrl, subscribeCallback).then(() => {
-          try {
-            addUser(nicknameParam);
-            /**
-             * Set the necessary variables for the waiting room page.
-             * The socket connection must be set before updating the page state,
-             * otherwise this whole method will re-run.
-             */
-            setNickname(nicknameParam);
-            setSocketConnected(true);
-            setPageState(2);
-            resolve();
-          } catch (err) {
-            reject(errorHandler(err.message));
-          }
-        }).catch((err) => {
-          reject(errorHandler(err.message));
-        });
-      }).catch((err) => {
-        reject(errorHandler(err.message));
-      });
-    });
+  // Get history object to be able to move between different pages
+  const history = useHistory();
 
   /**
-   * If the user is on the waiting room page state but not connected:
-   * add the user to the waiting room (which connects them to the socket).
-   * (This occurs when the create page redirects the user to the waiting page.)
+   * Redirect the user to the waiting room.
    */
-  if (!socketConnected && pageState === 2 && nickname) {
-    addUserToWaitingRoom(SOCKET_ENDPOINT, SUBSCRIBE_URL, nickname);
-  }
-
-  // Create variable to hold the "Join Page" content.
-  let joinPageContent: ReactElement | undefined;
-
-  switch (pageState) {
-    case 1:
-      // Render the "Enter nickname" state.
-      joinPageContent = (
-        <EnterNicknamePage
-          enterNicknameHeaderText="Enter a nickname to join the game!"
-          // Partial application of addUserToWaitingRoom function.
-          enterNicknameAction={
-            (nicknameParam: string) => addUserToWaitingRoom(SOCKET_ENDPOINT,
-              SUBSCRIBE_URL, nicknameParam)
-          }
-        />
-      );
-      break;
-    case 2:
-      // Render the Waiting room state.
-      joinPageContent = (
-        <div>
-          <LargeText>
-            You have entered the waiting room! Your nickname is &quot;
-            {nickname}
-            &quot;.
-          </LargeText>
-          { error ? <ErrorMessage message={error} /> : null }
-          <div>
-            {
-              users.map((user) => (
-                <UserNicknameText onClick={(event) => {
-                  deleteUser((event.target as HTMLElement).innerText);
-                }}
-                >
-                  {user.nickname}
-                </UserNicknameText>
-              ))
-            }
-          </div>
-        </div>
-      );
-      break;
-    default:
-  }
-
-  return joinPageContent;
+  const redirectToWaitingRoom = (nickname: string) => new Promise<undefined>((resolve) => {
+    history.push('/game/waiting', { nickname });
+    resolve();
+  });
+
+  // Render the "Enter nickname" state.
+  return (
+    <EnterNicknamePage
```

Comment:
I also got this error in the console from line 19 of this file: 
`Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.`

Reply:
Thanks for pointing that out, I think I saw that earlier; it isn't popping up anymore, so I think it was fixed by adding `useEffect(...)`.